pcdsdevices/mv_interface.py

```diff
@@ -19,9 +20,89 @@
 from . import utils as util
 
 logger = logging.getLogger(__name__)
+engineering_mode = True
 
 
-class MvInterface:
+class BaseInterface:
+    """
+    Interface layer to attach to any Device for SLAC features.
+
+    This class defines an API and some defaults for blacklisting tab-completion
+    results for new users to avoid confusion. The API involves setting the
+    following class variables on any subclasses:
+
+    tab_blacklist: list of string regex to omit from tab complete
+    tab_whitelist: list of string regex to include even if blacklisted by a
+                   parent class
+    """
+    tab_blacklist = ["OphydAttrList", "SUB_*", "_*", "actuate",
+                     "actuate_value", "clear_sub", "done", "done_value",
+                     "event_types", "get_device_tuple",
+                     "get_instantiated_signals", "hints", "kind", "root",
+                     "set", "setpoint", "stop_signal", "stop_value",
+                     "subscribe", "subscriptions", "trigger",
+                     "trigger_signals", "unsubscribe", "unsubscribe_all"]
+    tab_whitelist = ["__*__"]
```

Comment:
```suggestion
    tab_whitelist = ["__.*__"]
```

Reply:
I'm actually going to remove this entirely because I don't think the base class should have a whitelist, but thanks for reminding me that I have to do the regexes properly